tests/phpunit/api/v3/ValidateTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2016                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *  Tests for the generic validate API action.
+ *
+ * @package CiviCRM_APIv3
+ * @group headless
+ */
+class api_v3_ValidateTest extends CiviUnitTestCase {
+  /**
+   * This method is called before a test is executed.
+   */
+  protected function setUp() {
+    parent::setUp();
+  }
+
+  public function testEmptyContactValidate() {
+    $validation = $this->callAPISuccess('Contact', 'validate', array('action' => "create"));
+    $expectedOut = array(
+      'contact_type' => array(
+        'message' => "Mandatory key(s) missing from params array: contact_type",
+        'code' => "mandatory_missing",
+      ),
+    );
+    $this->assertEquals($validation['values'][0], $expectedOut);
```

Comment:
`$validation['values'][0]` - this nesting seems like a necessary evil, cannot return the output in this dictionary format:
```
    {
    "is_error": 0,
    "version": 3,
    "count": 1,
    "id": 0,
    "values": [
        {
            "contact_type": {
                "message": "Mandatory key(s) missing from params array: contact_type",
                "code": "mandatory_missing"
            }
        }
    ]
}
```